jupyterlab/federated_labextensions.py

```diff
@@ -359,40 +359,49 @@ def _get_labextension_metadata(module):
         Importable Python module exposing the
         magic-named `_jupyter_labextension_paths` function
     """
+
+    mod_path = osp.abspath(module)
+    if not osp.exists(mod_path):
+        raise FileNotFoundError('The path `{}` does not exist.'.format(module))
+
+    # Check if the path is a valid labextension
     try:
         m = importlib.import_module(module)
+        if hasattr(m, '_jupyter_labextension_paths') :
+            labexts = m._jupyter_labextension_paths()
+            return m, labexts
+        else :
+            m = None
+
     except Exception:
         m = None
 
-    if not hasattr(m, '_jupyter_labextension_paths'):
-        mod_path = osp.abspath(module)
-        if osp.exists(mod_path):
-
-            # Try getting the package name from setup.py
-            try:
-                package = subprocess.check_output([sys.executable, 'setup.py', '--name'], cwd=mod_path).decode('utf8').strip()
-            except subprocess.CalledProcessError:
-                from setuptools import find_packages
-                package = find_packages(mod_path)[0]
-
-            # Replace hyphens with underscores to match Python convention
-            package = package.replace('-', '_')
-
-            # Make sure the package is installed
-            import pkg_resources
-            try:
-                dist = pkg_resources.get_distribution(package)
-            except pkg_resources.DistributionNotFound:
-                subprocess.check_call([sys.executable, '-m', 'pip', 'install', '-e', mod_path])
-                sys.path.insert(0, mod_path)
-
-            m = importlib.import_module(package)
-
-    if not hasattr(m, '_jupyter_labextension_paths'):
-        raise KeyError('The Python module {} is not a valid labextension, '
-                       'it is missing the `_jupyter_labextension_paths()` method.'.format(module))
-    labexts = m._jupyter_labextension_paths()
-    return m, labexts
+    # Try getting the package name from setup.py
+    try:
+        package = subprocess.check_output([sys.executable, 'setup.py', '--name'], cwd=mod_path).decode('utf8').strip()
```

Comment:
We should maybe not even do this anymore. @maartenbreddels found that this was causing issues when the setup.py does some logging.